ab-testing/deploy-lambda/src/index.test.ts

```diff
@@ -0,0 +1,265 @@
+import type { CloudFormationCustomResourceEvent, Context } from "aws-lambda";
+
+describe("index", () => {
+	const mockContext: Context = {
+		callbackWaitsForEmptyEventLoop: false,
+		functionName: "test-function",
+		functionVersion: "$LATEST",
+		invokedFunctionArn:
+			"arn:aws:lambda:eu-west-1:123456789012:function:test-function",
+		memoryLimitInMB: "128",
+		awsRequestId: "test-aws-request-id",
+		logGroupName: "/aws/lambda/test-function",
+		logStreamName: "2025/12/01/[$LATEST]test-stream",
+		getRemainingTimeInMillis: () => 300000,
+		done: () => {},
+		fail: () => {},
+		succeed: () => {},
+	};
+
+	afterEach(() => {
+		mock.restoreAll();
+	});
+
+	describe("handler", async () => {
+		const customResourceSendMock = mock.fn();
+		const fetchAndDeployMock = mock.fn();
+
+		const getDictionaryItemsMock = mock.fn();
+		const updateDictionaryItemsMock = mock.fn(async () => {
+			return { status: "ok" };
+		});
+
+		const getServiceMock = mock.fn(async () => {
+			return {
+				getDictionary: async (name: string) => {
+					return {
+						name,
+						getItems: getDictionaryItemsMock,
+						updateItems: updateDictionaryItemsMock,
+					};
+				},
+			};
+		});
+
+		beforeEach(async () => {
+			mock.module("@guardian/ab-testing-config/lib/fastly/client.ts", {
+				namedExports: {
+					FastlyClient: class {
+						constructor() {}
+
+						getService = getServiceMock;
+					},
+				},
+			});
+
+			mock.module("./lib/fastly-config.ts", {
+				namedExports: {
+					getFastlyApiToken: async () => {
+						return "test-api-token";
+					},
+					getFastlyConfig: async () => {
+						return {
+							serviceId: "test-service-id",
+							serviceName: "test-service",
+							abTestsDictionaryName: "ab-tests",
+							mvtDictionaryName: "mvt-groups",
+						};
+					},
+				},
+			});
+
+			mock.module("./lib/custom-resource-response.ts", {
+				namedExports: {
+					send: customResourceSendMock,
+				},
+			});
+
+			mock.module("./lib/deploy.ts", {
+				namedExports: {
+					fetchAndDeployArtifacts: fetchAndDeployMock,
+				},
+			});
+
+			customResourceSendMock.mock.resetCalls();
+			fetchAndDeployMock.mock.resetCalls();
+			getDictionaryItemsMock.mock.resetCalls();
+			updateDictionaryItemsMock.mock.resetCalls();
+		});
+
+		await it("should send SUCCESS for Delete events without deploying", async () => {
+			const deleteEvent: CloudFormationCustomResourceEvent = {
+				RequestType: "Delete",
+				ServiceToken:
+					"arn:aws:lambda:eu-west-1:123456789012:function:test-function",
+				ResponseURL: "https://example.com/response",
+				StackId:
+					"arn:aws:cloudformation:eu-west-1:123456789012:stack/test-stack/test-id",
+				RequestId: "test-request-id",
+				LogicalResourceId: "TestResource",
+				ResourceType: "Custom::DictionaryDeployer",
+				PhysicalResourceId: "test-physical-id",
+				ResourceProperties: {
+					ServiceToken:
+						"arn:aws:lambda:eu-west-1:123456789012:function:test-function",
+				},
```

Comment:
minor: to reduce the file size, potentially split out the differences between the `Delete`, `Create` and `Update` events with an object spread per test?